File: examples/ruby/raw_send.rb
# Send raw WebSocket messages example
require_relative 'binary_options_tools_uni'

client = BinaryOptionsToolsUni::PocketOption.new("your-session-id")
sleep 5

# Subscribe to signals
client.raw_send('42["signals/subscribe"]')
puts "Sent signals subscription message"

# Subscribe to price updates
client.raw_send('42["price/subscribe"]')
puts "Sent price subscription message"

# Custom message example
custom_message = '42["custom/event",{"param":"value"}]'
client.raw_send(custom_message)
puts "Sent custom message: #{custom_message}"

# Send multiple messages in sequence
messages = [
  '42["chart/subscribe",{"asset":"EURUSD"}]',
  '42["trades/subscribe"]',
  '42["notifications/subscribe"]'
]

messages.each do |msg|
  client.raw_send(msg)
  puts "Sent message: #{msg}"
  sleep 1
end

File: examples/go/raw_send.go
// Example showing how to send raw WebSocket messages
package main

import (
	"fmt"
	"time"
	"binary_options_tools_uni"
)

func main() {
	client, err := binary_options_tools_uni.NewPocketOption("your-session-id")
	if err != nil {
		panic(err)
	}
	time.Sleep(5 * time.Second)

	// Create a validator that accepts all messages
	validator := binary_options_tools_uni.NewValidator()

	// Create a raw handler with no keep-alive message
	handler, err := client.CreateRawHandler(validator, nil)
	if err != nil {
		panic(err)
	}

	// Subscribe to signals
	err = handler.SendText("42[\"signals/subscribe\"]")
	if err != nil {
		fmt.Printf("Error sending signals subscription: %v\n", err)
	}
	fmt.Println("Sent signals subscription message")

	// Subscribe to price updates
	err = handler.SendText("42[\"price/subscribe\"]")
	if err != nil {
		fmt.Printf("Error sending price subscription: %v\n", err)
	}
	fmt.Println("Sent price subscription message")

	// Custom message example
	customMessage := "42[\"custom/event\",{\"param\":\"value\"}]"
	err = handler.SendText(customMessage)
	if err != nil {
		fmt.Printf("Error sending custom message: %v\n", err)
	}
	fmt.Printf("Sent custom message: %s\n", customMessage)
}
